### DISCUSSION TITLE
Docker action in same repository with Dockerfile in root

### DISCUSSION THREAD

**Discussion by raquelhortab** [community/community#26235] [Actions]
<p>I want to run a docker action that I created in my own repository (the same as the workflow), its path, relative to the repository root would be /.github/actions/automatic-tests. I also have a Dockerfile in the root of the repository, which I use to create a docker image that runs the tests in my rails app.</p>
<p>In the <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/metadata-syntax-for-github-actions#image" rel="nofollow noopener">documentation</a> it is said that _“To reference a Dockerfile local to your repository, use a path relative to the root of the repository” _and the example would be:</p>
<pre><code>image: 'Dockerfile'
</code></pre>
<p>However, it seems that, when running the action, the Dockerfile is searched in the actions folder, not in the root of the repository. The error is as follows:</p>
<pre><code>/home/runner/work/_actions/raquelhortab/hello_world_github_actions/master/.github/actions/automatic-tests/Dockerfile: no such file or directory
</code></pre>
<p>I cannot put the Dockerfile inside the actions folder as I need to COPY the whole app. The workaround would be to use git to clone the repo but I think this should not be the solution when working in the exact same repository…</p>
<p>Can anyone tell me what I am doing wrong or what I should do to make it work?</p>
<p>Thanks!</p>
<p>Raquel</p>

**Comment by raquelhortab** [comment, marked as answer]
<p>I finally got it to work with:</p>
<pre><code>image: '../../../Dockerfile'
</code></pre>
<p>So it seems the path is relative not to the repository but to the action root, I hope this helps someone!</p>

**Comment by francisfuzz** [comment]
<blockquote>
<p><a class="mention" href="/raquelhortab">@raquelhortab</a> wrote:</p>
</blockquote>
<blockquote>
<p>I finally got it to work with:</p>
</blockquote>
<blockquote>
<p>image: ‘…/…/…/Dockerfile’</p>
</blockquote>
<blockquote>
<p>So it seems the path is relative not to the repository but to the action root, I hope this helps someone!</p>
</blockquote>
<p>This helped me, thanks <a class="mention" href="/raquelhortab">@raquelhortab</a>! I see that the documentation was recently updated so the copy for <code>image</code> now says:</p>
<p>&gt;  <strong>Required for container actions</strong>  The Docker image to use as the container to run the action. The value can be the Docker base image name, a local <code>Dockerfile</code> in your repository, a public docker Hub or registry. To reference a <code>Dockerfile</code> local to your repository, use a path relative to your action metadata file. The application specified with the <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/metadata-syntax-for-github-actions#using" rel="nofollow noopener"><code>using</code></a> syntax will execute this file.</p>
